packages/shared/src/reparent-children.ts

```diff
@@ -0,0 +1,39 @@
+function restoreChildren(
+    placeholderItems: Comment[],
+    srcElements: Element[]
+): void {
+    if (srcElements.length != placeholderItems.length) {
```

Comment:
I think I can just remove this. This made me realize that we shouldn't collect the children to reparent on firstUpdate, but rather when you open the menu. Otherwise if the hierarchy changes we won't pick it up.